### PR TITLE
wine: gecko 2.47.1 -> 2.47.2 and add tests

### PR DESCRIPTION
**Pull request by chkno**
###### Motivation for this change
Wine complains about not having gecko & tries to fetch it over the internet because nixpkgs is providing the wrong version of gecko.  Fix this, and add a test so it's easy to notice in the future if the required & provided gecko versions come apart again.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

FYI: @SFrijters

### CODE REVIEW

**Comment by SFrijters** [comment]
Oh, good catch. Looks like this is needed for all 6.0+ versions, so this has been broken for a while.
I have never added anything to nix tests yet, so I don't know what is good/bad there. `hello.exe` is some test executable that requires `gecko` to work?

@avnik

**Comment by chkno** [comment]
hello.exe is [GNU Hello](https://www.gnu.org/software/hello/).  It prints "Hello, world!".  It doesn't require gecko, but wine notices that gecko is missing while it is initialising ~/.wine/ and mentions this.  A smarter, better wine might wait until gecko is actually needed before attempting to load it, which would allow this test to pass incorrectly.  So it's not a great test, but it's much better than nothing.

**Comment by avnik** [comment]
I have one  game, which using mshtml (and indirectly gecko via it), and it works with latest 6.9.
(old wineprefix)

File: nixos/tests/wine.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ system ? builtins.currentSystem, pkgs ? import ../.. {
+  inherit system;
+  config = { };
+}, }:
```
Comment (SuperSandro2000) [inline nixos/tests/wine.nix:4]: According to github's syntax highlighting something is broken here. Maybe the , is to much?

Comment (lovesegfault) [reply nixos/tests/wine.nix:4]: It works in practice, but you're right that it's not needed.

Making a PR fixing that.

Comment (lovesegfault) [reply nixos/tests/wine.nix:4]: https://github.com/NixOS/nixpkgs/pull/144665

**Pull request merged**